<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Olivia Gardner" />

<meta name="date" content="2020-03-23" />

<title>MultiDataAnalysis Tutorial - Merge Individual Files</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">MultiDataAnalysis Tutorial - Merge Individual Files</h1>
<h4 class="author">Olivia Gardner</h4>
<h4 class="date">2020-03-23</h4>



<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(MultiDataAnalysis)
<span class="co">#&gt; Loading required package: data.table</span></code></pre></div>
<p>This package facilitates combining data from multiple files, experiments, or datasets into one analysis. In particular, this package has two main functionalities: 1. Merging files that are split by sample into one data set by experiment/measurement type, and 2. Applying a regression model to all rows of one or more matrices using data from multiple data sets. In this tutorial, we will explore different applications and parameters for the function MergeIndividualFiles.</p>
<div id="merging-data-from-multiple-files-into-one-data-set" class="section level1">
<h1>Merging Data from Multiple Files into One Data Set</h1>
<p>The MergeIndividualFiles function reads in files containing data from individual samples and merges individual data into one dataset by variable or measurement.</p>
<p>For this example, we will use data that has been installed with the MultiDataAnalysis package in the extdata/ folder. You can also download these files directly from github: <a href="https://github.com/okg3/MultiDataAnalysis/tree/master/inst/extdata" class="uri">https://github.com/okg3/MultiDataAnalysis/tree/master/inst/extdata</a></p>
<p>We will be looking at sample data from an RNA editing experiment using 8 samples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># set dirPath to extdata directory in package path </span>
dirPath &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="kw">path.package</span>(<span class="st">&quot;MultiDataAnalysis&quot;</span>), <span class="st">&quot;/extdata/&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dirPath
<span class="co">#&gt; [1] &quot;C:/Users/ogard/Documents/R/win-library/3.5/MultiDataAnalysis/extdata/&quot;</span>
<span class="co"># Data is split into separate files by sample</span>
<span class="kw">list.files</span>(dirPath, <span class="dt">pattern =</span> <span class="st">&quot;.txt.gz&quot;</span>)
<span class="co">#&gt; [1] &quot;sample1.txt.gz&quot; &quot;sample2.txt.gz&quot; &quot;sample3.txt.gz&quot; &quot;sample4.txt.gz&quot;</span>
<span class="co">#&gt; [5] &quot;sample5.txt.gz&quot; &quot;sample6.txt.gz&quot; &quot;sample7.txt.gz&quot; &quot;sample8.txt.gz&quot;</span></code></pre></div>
<p>Letâ€™s load one file to look at values:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">file1 &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">paste0</span>(dirPath, <span class="st">&quot;sample1.txt.gz&quot;</span>))
<span class="kw">head</span>(file1)
<span class="co">#&gt;    Region  Position Reference Strand RepMask_feat RepMask_gid</span>
<span class="co">#&gt; 1:  chr10 100233036         T      1         SINE       AluJo</span>
<span class="co">#&gt; 2:  chr10 100233053         T      1         SINE       AluJo</span>
<span class="co">#&gt; 3:  chr10 101789717         T      1         SINE       AluJb</span>
<span class="co">#&gt; 4:  chr10 101789723         T      1         SINE       AluJb</span>
<span class="co">#&gt; 5:  chr10 101789782         T      1         SINE       AluJb</span>
<span class="co">#&gt; 6:  chr10 101789903         T      1         SINE       AluJb</span>
<span class="co">#&gt;             RefSeq_feat RefSeq_gid RADAR_feat RADAR_gid DARNED_feat DARNED_gid</span>
<span class="co">#&gt; 1: 3UTR,exon,transcript    CWF19L1       3UTR   CWF19L1           .    CWF19L1</span>
<span class="co">#&gt; 2: 3UTR,exon,transcript    CWF19L1       3UTR   CWF19L1           .    CWF19L1</span>
<span class="co">#&gt; 3:           transcript        OGA   intronic     MGEA5           .      MGEA5</span>
<span class="co">#&gt; 4:           transcript        OGA   intronic     MGEA5           .      MGEA5</span>
<span class="co">#&gt; 5:           transcript        OGA   intronic     MGEA5           .      MGEA5</span>
<span class="co">#&gt; 6:           transcript        OGA   intronic     MGEA5           .      MGEA5</span>
<span class="co">#&gt;    REDIportal_feat REDIportal_gid AllSubs Coverage-q25 MeanQ BaseCount[A,C,G,T]</span>
<span class="co">#&gt; 1:  ncRNA_intronic  RP11-316M21.6      TC           27 38.26      [0, 8, 0, 19]</span>
<span class="co">#&gt; 2:  ncRNA_intronic  RP11-316M21.6      TC           27 36.89      [0, 6, 0, 21]</span>
<span class="co">#&gt; 3:        intronic          MGEA5      TC           71 38.69     [0, 14, 0, 57]</span>
<span class="co">#&gt; 4:        intronic          MGEA5      TC           66 37.62     [0, 34, 0, 32]</span>
<span class="co">#&gt; 5:        intronic          MGEA5      TC           46 38.24     [1, 17, 0, 28]</span>
<span class="co">#&gt; 6:        intronic          MGEA5      TC           72 37.85     [0, 16, 0, 56]</span>
<span class="co">#&gt;    Frequency       Pvalue</span>
<span class="co">#&gt; 1:      0.30 1.215299e-02</span>
<span class="co">#&gt; 2:      0.22 5.014270e-02</span>
<span class="co">#&gt; 3:      0.20 2.631754e-04</span>
<span class="co">#&gt; 4:      0.52 4.167670e-12</span>
<span class="co">#&gt; 5:      0.38 1.355112e-05</span>
<span class="co">#&gt; 6:      0.22 5.983720e-05</span></code></pre></div>
<p>In this data, the unique editing events are identified by the columns Region, Position, Reference, and Strand.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">idCols &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Region&quot;</span>, <span class="st">&quot;Position&quot;</span>, <span class="st">&quot;Reference&quot;</span>, <span class="st">&quot;Strand&quot;</span>)</code></pre></div>
<p>The columns AllSubs, Coverage-q25, MeanQ, BaseCount[A,C,G,T], Frequency, and Pvalue all represent individual meansurements that are specific to this sample. In order to compare measurements across samples, we will want to combine all of the Frequency columns from different samples into one matrix, all Coverage columns into another matrix, etc.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">IndVars &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;AllSubs&quot;</span>, <span class="st">&quot;Coverage-q25&quot;</span>, <span class="st">&quot;MeanQ&quot;</span>, <span class="st">&quot;BaseCount[A,C,G,T]&quot;</span>, 
             <span class="st">&quot;Frequency&quot;</span>, <span class="st">&quot;Pvalue&quot;</span>)</code></pre></div>
<p>The remaining columns are annotations of the unique editing events which we will want to keep in a separate annotation data.frame.</p>
<p>To combine these datasets, we will use the MergeIndividualFiles function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">out &lt;-<span class="st"> </span><span class="kw">MergeIndividualFiles</span>(
  <span class="dt">fileDirectory =</span> dirPath,
  <span class="dt">filePattern =</span> <span class="st">&quot;.txt.gz&quot;</span>, 
  <span class="dt">indVars =</span> IndVars,
  <span class="dt">IDcol =</span> idCols,
  <span class="dt">na.strings =</span> <span class="st">&quot;-&quot;</span>
)
<span class="kw">names</span>(out)
<span class="co">#&gt; [1] &quot;Annotation&quot;         &quot;AllSubs&quot;            &quot;Coverage-q25&quot;      </span>
<span class="co">#&gt; [4] &quot;MeanQ&quot;              &quot;BaseCount[A,C,G,T]&quot; &quot;Frequency&quot;         </span>
<span class="co">#&gt; [7] &quot;Pvalue&quot;</span></code></pre></div>
<p>Our merged dataset is a list containing one Annotation data.frame and 6 matrices with individual sample values for the 6 variables defined in indVars.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(out<span class="op">$</span>Annotation)
<span class="co">#&gt;                     id Region  Position Reference Strand RepMask_feat</span>
<span class="co">#&gt; 1: chr10_100233036_T_1  chr10 100233036         T      1         SINE</span>
<span class="co">#&gt; 2: chr10_100233053_T_1  chr10 100233053         T      1         SINE</span>
<span class="co">#&gt; 3: chr10_101789717_T_1  chr10 101789717         T      1         SINE</span>
<span class="co">#&gt; 4: chr10_101789723_T_1  chr10 101789723         T      1         SINE</span>
<span class="co">#&gt; 5: chr10_101789782_T_1  chr10 101789782         T      1         SINE</span>
<span class="co">#&gt; 6: chr10_101789903_T_1  chr10 101789903         T      1         SINE</span>
<span class="co">#&gt;    RepMask_gid          RefSeq_feat RefSeq_gid RADAR_feat RADAR_gid DARNED_feat</span>
<span class="co">#&gt; 1:       AluJo 3UTR,exon,transcript    CWF19L1       3UTR   CWF19L1           .</span>
<span class="co">#&gt; 2:       AluJo 3UTR,exon,transcript    CWF19L1       3UTR   CWF19L1           .</span>
<span class="co">#&gt; 3:       AluJb           transcript        OGA   intronic     MGEA5           .</span>
<span class="co">#&gt; 4:       AluJb           transcript        OGA   intronic     MGEA5           .</span>
<span class="co">#&gt; 5:       AluJb           transcript        OGA   intronic     MGEA5           .</span>
<span class="co">#&gt; 6:       AluJb           transcript        OGA   intronic     MGEA5           .</span>
<span class="co">#&gt;    DARNED_gid REDIportal_feat REDIportal_gid</span>
<span class="co">#&gt; 1:    CWF19L1  ncRNA_intronic  RP11-316M21.6</span>
<span class="co">#&gt; 2:    CWF19L1  ncRNA_intronic  RP11-316M21.6</span>
<span class="co">#&gt; 3:      MGEA5        intronic          MGEA5</span>
<span class="co">#&gt; 4:      MGEA5        intronic          MGEA5</span>
<span class="co">#&gt; 5:      MGEA5        intronic          MGEA5</span>
<span class="co">#&gt; 6:      MGEA5        intronic          MGEA5</span>

<span class="kw">head</span>(out<span class="op">$</span>Frequency)
<span class="co">#&gt;                     sample1 sample2 sample3 sample4 sample5 sample6 sample7</span>
<span class="co">#&gt; chr10_100233036_T_1    0.30    0.33    0.26    0.30    0.22    0.25    0.23</span>
<span class="co">#&gt; chr10_100233053_T_1    0.22    0.47    0.25    0.50    0.30    0.52    0.38</span>
<span class="co">#&gt; chr10_101789717_T_1    0.20    0.19    0.21    0.29    0.35    0.26    0.30</span>
<span class="co">#&gt; chr10_101789723_T_1    0.52    0.54    0.48    0.53    0.66    0.71    0.56</span>
<span class="co">#&gt; chr10_101789782_T_1    0.38    0.43    0.41    0.33    0.49    0.62    0.40</span>
<span class="co">#&gt; chr10_101789903_T_1    0.22    0.29    0.38    0.30    0.41    0.28    0.28</span>
<span class="co">#&gt;                     sample8</span>
<span class="co">#&gt; chr10_100233036_T_1    0.29</span>
<span class="co">#&gt; chr10_100233053_T_1    0.32</span>
<span class="co">#&gt; chr10_101789717_T_1    0.20</span>
<span class="co">#&gt; chr10_101789723_T_1    0.57</span>
<span class="co">#&gt; chr10_101789782_T_1    0.55</span>
<span class="co">#&gt; chr10_101789903_T_1    0.34</span></code></pre></div>
<p>Each matrix has 8 columns names after the file names, after removing the pattern defined in filePattern. Matrix rownames match the id column in the Annotation data.frame.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
